wls-common/src/main/java/org/jboss/arquillian/container/wls/ShrinkWrapUtil.java

```diff
@@ -106,22 +112,93 @@ public static URL toURL(final Descriptor descriptor)
     */
    public static File toFile(final Archive<?> archive)
    {
-      // create a random named temp file, then delete and use it as a directory
-      try
-      {
-         File root = File.createTempFile("arquillian", archive.getName());
-         root.delete();
-         root.mkdirs();
-         
-         File deployment = new File(root, archive.getName());
-         deployment.deleteOnExit();
-         archive.as(ZipExporter.class).exportTo(deployment, true);
-         return deployment;
-      }
-      catch (Exception e) 
+      return toFile(archive, false);
+   }
+
+   /**
+    * Creates a tmp folder and exports the file. Returns the URL for that file location.
+    *
+    * @param archive Archive to export
+    * @param exploded Specifies, whether to explode the archive after creation
+    * @return
+    */
+   public static File toFile(final Archive<?> archive, final boolean exploded)
+   {
+       // create a random named temp file, then delete and use it as a directory
+       try
+       {
+           File root = File.createTempFile("arquillian", archive.getName());
+           root.delete();
+           root.mkdirs();
+
+           File deployment = new File(root, archive.getName());
+           deployment.deleteOnExit();
+           archive.as(ZipExporter.class).exportTo(deployment, true);

```

Comment:
Wouldn't this export it as a Compressed file even if requested to export it exploded?